packages/page-accounts/src/modals/Create.tsx

```diff
@@ -134,20 +141,33 @@ function generateSeed (_seed: string | undefined | null, derivePath: string, see
 }
 
 function updateAddress (seed: string, derivePath: string, seedType: SeedType, pairType: PairType): AddressState {
-  const deriveValidation = deriveValidate(seed, seedType, derivePath, pairType);
-
-  let isSeedValid = seedType === 'raw'
-    ? rawValidate(seed)
-    : mnemonicValidate(seed);
   let address: string | null = null;
+  let isSeedValid = true;
+  let deriveValidation: DeriveValidationOutput = {};
 
-  if (!deriveValidation?.error && isSeedValid) {
+  if (pairType === 'ethereum') {
     try {
       address = addressFromSeed(seed, derivePath, pairType);
     } catch (error) {
       console.error(error);
 
       isSeedValid = false;
+      deriveValidation = { error: (error as Error).message ? (error as Error).message : (error as Error).toString() };
+    }
+  } else {
+    deriveValidation = deriveValidate(seed, seedType, derivePath, pairType);
+    isSeedValid = seedType === 'raw'
```

Comment:
Not sure at all why the complete movement here.

For all types, 

- the path needs validation - For ethereum, you would use the `hdValidatePath`, for the rest as-is.
- You need to validate the seeds in all cases for validity, not ignore in the case of Ethereum
- then the creation happens

So would put the original flow back and only have the specific path validation parts different here.

Reply:
>You need to validate the seeds in all cases for validity, not ignore in the case of Ethereum

I don't understand that part. deriveValidate used keyExtractSuri. what verification is keyExtractSuri doing?

Reply:
So that means deriveValidate should just, in the case of an `ethereum` pairType, use the `hdValidatePath` - sadly only true/false, but that is good enough as a start.